src/numerical_checks.jl

```diff
@@ -82,6 +82,98 @@ function check_inverse_retraction(
+
+You can provide an `inverse_retraction_method` and `vector_transport_method` to be used,
+but the verification will then be less accurate.
+
+The tests performed are the following based on sampling the geodesic ``\gamma(t) = ``[`geodesic`](@ref)`(M, p, X)` at
+`N` equidistant points ``t_i = i/(N-1)``, `i=0,...,N-1` we denote by ``p_i = \gamma(t_i)``
+
+1. Compute the norms of the tangent vectors ``\dot\gamma(t_i) ≈ X_i = \log_{p_i}(p_{i+1})``, ``i=0,...,N-2``
+   and check how far these norms deviate from their mean value.
+2. Check how far this mean deviates from being ``\frac{||X||}{N-1}``
+3. Check that the parallel transport of each ``X_i`` from ``p_i`` to ``p_{i+1}`` is close to ``X_{i+1}``.
+
+# Arguments
+* `M`:    the manifold to check
+* `p`:    point on the manifold to start the geodesic, a random point is used by default
+* `X`:    tangent vector at `p` to start the geodesic, a random tangent vector is used by default
+
+# Keyword arguments
+* `tol`:     if all errors are below this tolerance, the geodesic is considered to be exact
+* `io`:      provide an `IO` to print the result to
+* `N`:       number of points to sample the geodesic on ``[0,1]`` (default: 101)
+* `plot`:    whether to plot the result (if `Plots.jl` is loaded).
+* `error`:   specify how to report errors: `:none`, `:info`, `:warn`, or `:error` are available
+* `inverse_retraction_method`:  method to use for the inverse retraction, it is recommended to use [`LogarithmicInverseRetraction`](@ref)
+* `vector_transport_method`:    method to use for the vector transport, it is recommended to use [`ParallelTransport`](@ref)
+"""
+function check_geodesic(
+        M::AbstractManifold,
+        p = rand(M),
+        X = rand(M; vector_at = p);
+        tol::Real = 1.0e-12,
+        io::Union{IO, Nothing} = nothing,
+        N::Int = 101,
+        plot::Bool = false,
+        error::Symbol = :none,
+        inverse_retraction_method::AbstractInverseRetractionMethod = default_inverse_retraction_method(M),
+        vector_transport_method::AbstractVectorTransportMethod = default_vector_transport_method(M),
+    )
+    T = range(0.0, 1.0; length = N)
+    γ = geodesic(M, p, X)
+    # points `p_i` to evaluate the error function at
+    ps = [ γ(t) for t in T ]
+    Xs = [ inverse_retract(M, ps[i], ps[i + 1], inverse_retraction_method) for i in 1:(N - 1) ]
+    norms = [ norm(M, p, X) for (p, X) in zip(ps[1:(end - 1)], Xs) ]
+    mean_norm = sum(norms) / length(norms)
+    # errors_1 how far awary are we from constant speed
+    errors_norm = [ abs(n - mean_norm) for n in norms ]
+    err_n_max = maximum(errors_norm)
+    # PT Xs to their neighbors, now Xi and Yi are in the same tangent space
+    Ys = [ vector_transport_to(M, ps[i + 1], Xs[i + 1], ps[i], vector_transport_method) for i in 1:(N - 2)]
+    errors_pt = [ norm(M, p, X - Y) for (p, X, Y) in zip(ps[1:(N - 2)], Xs[1:(N - 2)], Ys) ]
+    err_pt_max = maximum(errors_pt)
```

Comment:
I think maybe on the plot, it's possible to explain a bit better what is going on. The interpretation of the `errors_norm` is clear for me we want a constant speed. The angle check is also clear.

`errors_pt` is, if I am not mistaken, approximation to the covariant acceleration, which should be 0 if it's the geodesic. Am I right?

Reply:
If we PT X_i+1 to p_i (called Y_i then,
we should have that they are basically equal. 

One check is that ||X_i - Y_i || = 0
and another good check is that their angle is zero. That means that the cos of the angle is 1, and that is the inner divided by both norms. That is the third check.

So yes e_n and E_\alpha are a bit redundant but illustrate the same aspect. I can think about how to best document that.